src/anm/animation/tween.js

```diff
@@ -27,152 +34,284 @@ var errors = require('../errors.js');
+ * * `elm.modify(Tween.rotate().from(0).to(Math.PI / 2).band(0, 2).easing('in'))`
+ * * `elm.modify(Tween.translate().from(0, 0).to(100, 100))`
+ * * `elm.modify(Tween.translate().value('M0 0 100 100'))`
+ * * `elm.modify(Tween.translate().value(new Path().move(0, 0).line(100, 100)))`
+ * * `elm.modify(Tween.rotatetopath())`
+ * * `elm.modify(Tween['rotatetopath']())`
  */
-function Tween(tween_type, data) {
-    if (!tween_type) throw errors.element('Tween type is required to be specified or function passed');
-    var func;
-    if (is.fun(tween_type)) {
-        func = tween_type;
-    } else {
-        func = Tweens[tween_type](data);
-        func.tween = tween_type;
-    }
-    func.is_tween = true;
-    var mod = Modifier(func, C.MOD_TWEEN);
-    mod.$data = data;
-    // FIXME: value should be an array i.e. for scale tween, use object like { sx: <num>, sy: <num> } instead
-    mod.from = function(val) {
-                   if (!is.defined(val) && this.$data) return this.$data[0];
-                   if (!this.$data) this.$data = [];
-                   this.$data[0] = val;
-                   return this;
-               };
-    mod.to   = function(val) {
-                   if (!is.defined(val) && this.$data) return this.$data[1];
-                   if (!this.$data) this.$data = [];
-                   this.$data[1] = val;
-                   return this;
-               };
-    mod.data = data_block_fn; // FIXME
+var Tween = Modifier;
+
+function createTween(type, value) {
+    var mod = new Modifier(null, C.MOD_TWEEN); // mod.func will be set to `null` first
+    var def = Registry[type];
+    mod.def = def;
+    mod.func = function(t, dt, duration) { if (mod.$tween) mod.$tween.call(this, t, dt, duration); };
+    mod.is_tween = true;
+    if (is.defined(value)) mod.value(value);
     return mod;
 }
 
-var data_block_fn = function() {
-    throw errors.element("Data should be passed to tween in a constructor or using from()/to() methods");
-};
+// tween type priority depends on the order the types were registered
+
+var Registry = {};
 
-// TODO: add function to add every tween type in easy way, may be separate module?
-// .tween(new anm.Tween(C.T_TRANSLATE, [[0, 0], [100, 100]]).band(0, Infinity)) does not work
+Tween.DEFAULT_FROM = function(_from, prev) { return is.defined(prev) ? [ _from,   prev[1] ] : [ _from, null ]; };
+Tween.DEFAULT_TO   = function(to,    prev) { return is.defined(to)   ? [ prev[0], to ]      : [  null,   to ]; };
 
-// tween order
-Tween.TWEENS_PRIORITY = {};
-Tween.TWEENS_COUNT = 0;
+Tween.register = function(type, definition) {
+    definition = is.fun(definition) ? { func: definition } : definition;
+    definition.from = definition.from || Tween.DEFAULT_FROM;
+    definition.to   = definition.to   || Tween.DEFAULT_TO;
+    Registry[type] = definition;
+    Tween[type] = function(value) { return createTween(type, value); };
+}
 
-var Tweens = {};
+Tween._$ = createTween;
```

Comment:
Probably it's not needed to be exposed. It was planned to act similar to `Element._$`, which allows user to do something like `var element = Element._$` and then do nice chaining like: `element(<name>).band().fill()...`. So user decides what name should the chain root have.